libdnf5-cli/output/advisorylist.cpp

```diff
@@ -104,6 +106,70 @@ void print_advisorylist_table(
     scols_unref_table(table);
 }
 
+static json_object *adv_pkg_as_json(
+    auto &adv_pkg) {
+    auto advisory = adv_pkg->get_advisory();
+    json_object * json_advisory = json_object_new_object();
+    json_object_object_add(json_advisory, "name",
+			   json_object_new_string(advisory->get_name().c_str()));
+    json_object_object_add(json_advisory, "type",
+			   json_object_new_string(advisory->get_type().c_str()));
+    json_object_object_add(json_advisory, "severity",
+			   json_object_new_string(advisory->get_severity().c_str()));
+    json_object_object_add(json_advisory, "nevra",
+			   json_object_new_string(adv_pkg->get_nevra().c_str()));
+    json_object_object_add(json_advisory, "from",
+			   json_object_new_string(advisory->get_vendor().c_str()));
+    json_object_object_add(json_advisory, "status",
+			   json_object_new_string(advisory->get_status().c_str()));
+    json_object_object_add(json_advisory, "type",
+			   json_object_new_string(advisory->get_type().c_str()));
+    json_object_object_add(json_advisory, "title",
+			   json_object_new_string(advisory->get_title().c_str()));
+    json_object_object_add(json_advisory, "description",
+			   json_object_new_string(advisory->get_description().c_str()));
+    std::string message = advisory->get_message();
+    if (message.size() > 0)
+        json_object_object_add(json_advisory, "message",
+			       json_object_new_string(message.c_str()));
+    return json_advisory;
+}
+
+static json_object * adv_refs_as_json(
+    auto &advisory) {
+    json_object * json_adv_references = json_object_new_array();
+    auto references = advisory.get_references();
+    for (auto reference : references) {
+        json_object * json_reference = json_object_new_object();
+	json_object_object_add(json_reference,
+			       "id", json_object_new_string(reference.get_id().c_str()));
+	json_object_object_add(json_reference,
+			       "type", json_object_new_string(reference.get_type().c_str()));
+	json_object_object_add(json_reference,
+			       "title", json_object_new_string(reference.get_title().c_str()));
+	json_object_object_add(json_reference,
+			       "url", json_object_new_string(reference.get_url().c_str()));
+        json_object_array_add(json_adv_references, json_reference);
```

Comment:
List of references like this will be part of the `info --json` command output, here as well I would keep the information and its layout identical to the plain text command.

Reply:
Updated to be much closer to the text output. I added the advisory id / name to the reference output (`--with-bz`/`--with-cve`) to make it easier to then pump these into `advisory info`, or to only pump the ones you don't already know about into it.